urweb does not build with clang

[Building with clang gives](https://travis-ci.org/JasonGross/urweb/jobs/99487440#L343)
```
/bin/bash ../../libtool --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I../../include/urweb  -I./../../include/urweb   -Wimplicit -Wall -Werror -Wno-format-security -Wno-deprecated-declarations -U_FORTIFY_SOURCE  -g -O2 -MT urweb.lo -MD -MP -MF .deps/urweb.Tpo -c -o urweb.lo urweb.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I../../include/urweb -I./../../include/urweb -Wimplicit -Wall -Werror -Wno-format-security -Wno-deprecated-declarations -U_FORTIFY_SOURCE -g -O2 -MT urweb.lo -MD -MP -MF .deps/urweb.Tpo -c urweb.c  -fPIC -DPIC -o .libs/urweb.o
urweb.c:4570:3: error: redefinition of
      typedef 'uw_Sqlcache_Entry' is a
      C11 feature
      [-Werror,-Wtypedef-redefinition]
} uw_Sqlcache_Entry;
  ^
../../include/urweb/types_cpp.h:133:34: note: 
      previous definition is here
typedef struct uw_Sqlcache_Entry uw_Sqlc...
                                 ^
1 error generated.
make[1]: *** [urweb.lo] Error 1
make[1]: Leaving directory `/home/travis/build/JasonGross/urweb/src/c'
make: *** [all-recursive] Error 1
```

#5 fixes `make`, but [`make test` still fails with](https://travis-ci.org/JasonGross/urweb/jobs/99502414#L428)
```
bin/urweb -boot -noEmacs -dbms sqlite -db /tmp/urweb.db -demo /Demo demo
/usr/lib/gcc/x86_64-linux-gnu/4.6/../../../x86_64-linux-gnu/libcrypto.a(dso_dlfcn.o): In function `dlfcn_globallookup':
(.text+0x21): warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/lib/gcc/x86_64-linux-gnu/4.6/../../../x86_64-linux-gnu/libsqlite3.a(sqlite3.o): In function `pthreadMutexTry':
(.text+0x7295): undefined reference to `pthread_mutex_trylock'
/usr/lib/gcc/x86_64-linux-gnu/4.6/../../../x86_64-linux-gnu/libsqlite3.a(sqlite3.o): In function `pthreadMutexFree':
(.text+0x9385): undefined reference to `pthread_mutex_destroy'
/usr/lib/gcc/x86_64-linux-gnu/4.6/../../../x86_64-linux-gnu/libsqlite3.a(sqlite3.o): In function `pthreadMutexAlloc':
(.text+0x14091): undefined reference to `pthread_mutex_init'
/usr/lib/gcc/x86_64-linux-gnu/4.6/../../../x86_64-linux-gnu/libsqlite3.a(sqlite3.o): In function `pthreadMutexAlloc':
(.text+0x140e4): undefined reference to `pthread_mutexattr_init'
/usr/lib/gcc/x86_64-linux-gnu/4.6/../../../x86_64-linux-gnu/libsqlite3.a(sqlite3.o): In function `pthreadMutexAlloc':
(.text+0x140f1): undefined reference to `pthread_mutexattr_settype'
/usr/lib/gcc/x86_64-linux-gnu/4.6/../../../x86_64-linux-gnu/libsqlite3.a(sqlite3.o): In function `pthreadMutexAlloc':
(.text+0x140fc): undefined reference to `pthread_mutex_init'
/usr/lib/gcc/x86_64-linux-gnu/4.6/../../../x86_64-linux-gnu/libsqlite3.a(sqlite3.o): In function `pthreadMutexAlloc':
(.text+0x14104): undefined reference to `pthread_mutexattr_destroy'
/usr/lib/gcc/x86_64-linux-gnu/4.6/../../../x86_64-linux-gnu/libsqlite3.a(sqlite3.o): In function `pthreadMutexLeave':
(.text+0x7281): undefined reference to `pthread_mutex_unlock'
/usr/lib/gcc/x86_64-linux-gnu/4.6/../../../x86_64-linux-gnu/libsqlite3.a(sqlite3.o): In function `pthreadMutexEnter':
(.text+0x72b1): undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(openssl.o): In function `uw_init_crypto':
/home/travis/build/JasonGross/urweb/src/c/openssl.c:72: undefined reference to `pthread_mutex_init'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(openssl.o): In function `thread_id':
/home/travis/build/JasonGross/urweb/src/c/openssl.c:43: undefined reference to `pthread_self'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(openssl.o): In function `lock_or_unlock':
/home/travis/build/JasonGross/urweb/src/c/openssl.c:50: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/openssl.c:55: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `find_client':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:245: undefined reference to `pthread_self'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:247: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:256: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `uw_client_connect':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:286: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `find_client':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:250: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `uw_client_connect':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:289: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:296: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:322: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `find_client':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:245: undefined reference to `pthread_self'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:247: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:256: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `use_client':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:226: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:228: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:229: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `new_client':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:186: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `find_client':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:250: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `new_client':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:200: undefined reference to `pthread_mutex_init'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:201: undefined reference to `pthread_mutex_init'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:206: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:215: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:220: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `use_client':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:226: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:228: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:229: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `new_client':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:193: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `release_client':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:233: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:234: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:236: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `uw_commit':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:3431: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:3438: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `release_client':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:233: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:234: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:236: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `find_client':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:245: undefined reference to `pthread_self'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:247: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:256: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `client_send':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:339: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:355: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `uw_commit':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:3502: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `release_client':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:233: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:234: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:236: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `uw_commit':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:3474: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `release_client':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:233: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:234: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:236: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `find_client':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:250: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `uw_prune_clients':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:3650: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:3651: undefined reference to `pthread_self'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:3656: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:3676: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:3679: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `uw_Sqlcache_check':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:4651: undefined reference to `pthread_rwlock_rdlock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:4649: undefined reference to `pthread_rwlock_wrlock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:4680: undefined reference to `pthread_rwlock_unlock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:4662: undefined reference to `pthread_rwlock_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `uw_Sqlcache_rlock':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:4785: undefined reference to `pthread_rwlock_rdlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `uw_Sqlcache_wlock':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:4790: undefined reference to `pthread_rwlock_wrlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `uw_Sqlcache_store':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:4810: undefined reference to `pthread_rwlock_rdlock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:4812: undefined reference to `pthread_rwlock_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `uw_Sqlcache_flush':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:4825: undefined reference to `pthread_rwlock_wrlock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:4862: undefined reference to `pthread_rwlock_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `uw_Sqlcache_storeCommitOne':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:4689: undefined reference to `pthread_rwlock_wrlock'
/home/travis/build/JasonGross/urweb/src/c/urweb.c:4725: undefined reference to `pthread_rwlock_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(urweb.o): In function `uw_Sqlcache_free':
/home/travis/build/JasonGross/urweb/src/c/urweb.c:4765: undefined reference to `pthread_rwlock_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(request.o): In function `pthread_create_big':
/home/travis/build/JasonGross/urweb/src/c/request.c:135: undefined reference to `pthread_attr_init'
/home/travis/build/JasonGross/urweb/src/c/request.c:138: undefined reference to `pthread_attr_setstacksize'
/home/travis/build/JasonGross/urweb/src/c/request.c:143: undefined reference to `pthread_create'
/home/travis/build/JasonGross/urweb/src/c/request.c:135: undefined reference to `pthread_attr_init'
/home/travis/build/JasonGross/urweb/src/c/request.c:138: undefined reference to `pthread_attr_setstacksize'
/home/travis/build/JasonGross/urweb/src/c/request.c:143: undefined reference to `pthread_create'
/home/travis/build/JasonGross/urweb/src/c/request.c:135: undefined reference to `pthread_attr_init'
/home/travis/build/JasonGross/urweb/src/c/request.c:138: undefined reference to `pthread_attr_setstacksize'
/home/travis/build/JasonGross/urweb/src/c/request.c:143: undefined reference to `pthread_create'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(queue.o): In function `uw_dequeue':
/home/travis/build/JasonGross/urweb/src/c/queue.c:49: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/queue.c:51: undefined reference to `pthread_cond_wait'
/home/travis/build/JasonGross/urweb/src/c/queue.c:53: undefined reference to `pthread_mutex_unlock'
/home/travis/build/JasonGross/urweb/src/c/.libs/liburweb.a(queue.o): In function `uw_enqueue':
/home/travis/build/JasonGross/urweb/src/c/queue.c:59: undefined reference to `pthread_mutex_lock'
/home/travis/build/JasonGross/urweb/src/c/queue.c:61: undefined reference to `pthread_cond_broadcast'
/home/travis/build/JasonGross/urweb/src/c/queue.c:62: undefined reference to `pthread_mutex_unlock'
/usr/lib/gcc/x86_64-linux-gnu/4.6/libgcc_eh.a(unwind-dw2-fde-glibc.o): In function `__register_frame_info_bases':
(.text+0x16c4): undefined reference to `pthread_mutex_lock'
/usr/lib/gcc/x86_64-linux-gnu/4.6/libgcc_eh.a(unwind-dw2-fde-glibc.o): In function `__register_frame_info_table_bases':
(.text+0x178b): undefined reference to `pthread_mutex_lock'
/usr/lib/gcc/x86_64-linux-gnu/4.6/libgcc_eh.a(unwind-dw2-fde-glibc.o): In function `__deregister_frame_info_bases':
(.text+0x182e): undefined reference to `pthread_mutex_lock'
/usr/lib/gcc/x86_64-linux-gnu/4.6/libgcc_eh.a(unwind-dw2-fde-glibc.o): In function `__deregister_frame_info_bases':
(.text+0x18b6): undefined reference to `pthread_mutex_unlock'
/usr/lib/gcc/x86_64-linux-gnu/4.6/libgcc_eh.a(unwind-dw2-fde-glibc.o): In function `_Unwind_Find_FDE':
(.text+0x1966): undefined reference to `pthread_mutex_lock'
/usr/lib/gcc/x86_64-linux-gnu/4.6/libgcc_eh.a(unwind-dw2-fde-glibc.o): In function `_Unwind_Find_FDE':
(.text+0x19b7): undefined reference to `pthread_mutex_unlock'
/usr/lib/gcc/x86_64-linux-gnu/4.6/libgcc_eh.a(unwind-dw2-fde-glibc.o): In function `__register_frame_info_bases':
(.text+0x16e3): undefined reference to `pthread_mutex_unlock'
/usr/lib/gcc/x86_64-linux-gnu/4.6/libgcc_eh.a(unwind-dw2-fde-glibc.o): In function `__register_frame_info_table_bases':
(.text+0x17aa): undefined reference to `pthread_mutex_unlock'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [test] Error 1
```

--- a/include/urweb/types_cpp.h
+++ b/include/urweb/types_cpp.h
@@ -130,7 +130,12 @@ typedef struct uw_Sqlcache_Value {
   unsigned long timeValid;
 } uw_Sqlcache_Value;
 
-typedef struct uw_Sqlcache_Entry uw_Sqlcache_Entry;
+typedef struct uw_Sqlcache_Entry {
+  char *key;
+  uw_Sqlcache_Value *value;
+  unsigned long timeInvalid;
+  UT_hash_handle hh;
+} uw_Sqlcache_Entry;
 
 typedef struct uw_Sqlcache_Cache {
   pthread_rwlock_t lockOut;

--- a/src/c/urweb.c
+++ b/src/c/urweb.c
@@ -4562,13 +4562,6 @@ void uw_set_remoteSock(uw_context ctx, int sock) {
 
 // Sqlcache
 
-typedef struct uw_Sqlcache_Entry {
-  char *key;
-  uw_Sqlcache_Value *value;
-  unsigned long timeInvalid;
-  UT_hash_handle hh;
-} uw_Sqlcache_Entry;
-
 static void uw_Sqlcache_freeValue(uw_Sqlcache_Value *value) {
   if (value) {
     free(value->result);

